datalad/core/local/diff.py

```diff
@@ -0,0 +1,335 @@
+            # distinguish rsync-link syntax to identify
+            # a dataset as whole (e.g. 'ds') vs its
+            # content (e.g. 'ds/')
+            # special case is the root dataset, always report its content
+            # changes
+            orig_path = text_type(p)
+            resolved_path = rev_resolve_path(p, dataset)
+            p = \
+                resolved_path, \
+                orig_path.endswith(op.sep) or resolved_path == ds.pathobj
+            str_path = text_type(p[0])
+            root = rev_get_dataset_root(str_path)
+            if root is None:
+                # no root, not possibly underneath the refds
+                yield dict(
+                    action='status',
+                    path=str_path,
+                    refds=ds.path,
+                    status='error',
+                    message='path not underneath this dataset',
+                    logger=lgr)
+                continue
+            if path_under_rev_dataset(ds, str_path) is None:
+                # nothing we support handling any further
+                # there is only a single refds
+                yield dict(
+                    path=str_path,
+                    refds=ds.path,
+                    action='diff',
+                    status='error',
+                    message=(
+                        "dataset containing given paths is not underneath "
+                        "the reference dataset %s: %s",
+                        ds, str_path),
+                    logger=lgr,
+                )
+                continue
+
+            ps.append(p)
+        path = ps
+
+    # TODO we might want to move away from the single-pass+immediate-yield
+    # paradigm for this command. If we gather all information first, we
+    # could do post-processing and detect when a file (same gitsha, or same
+    # key) was copied/moved from another dataset. Another command (e.g.
+    # rev-save) could act on this information and also move/copy
+    # availability information or at least enhance the respective commit
+    # message with cross-dataset provenance info
+
+    # cache to help avoid duplicate status queries
+    content_info_cache = {}
+    for res in _diff_ds(
+            ds,
+            fr,
+            to,
+            constant_refs,
+            recursion_limit
+            if recursion_limit is not None and recursive
+            else -1 if recursive else 0,
+            # TODO recode paths to repo path reference
+            origpaths=None if not path else OrderedDict(path),
+            untracked=untracked,
+            annexinfo=annex,
+            eval_file_type=True,
+            cache=content_info_cache):
+        res.update(
+            refds=ds.path,
+            logger=lgr,
+            action='diff',
+        )
+        yield res
+
+
+def _diff_ds(ds, fr, to, constant_refs, recursion_level, origpaths, untracked,
+             annexinfo, eval_file_type, cache):
+    if not ds.repo:
```

Comment:
```suggestion
    if not ds.is_installed():
```

Reply:
Is it faster?

Reply:
Speed seems to be exactly the same.

Reply:
Adjusted.

Reply:
my comment was not about speed but about coded semantic/assumptions -- are you really care about `.repo` being there or dataset being installed (whatever that means and regardless how it is assessed)?  I thought that it about it being installed.